packages/plugins/plugin-sheet/src/integrations/thread-ranges.ts

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type MutableRefObject, useCallback, useEffect, useMemo } from 'react';
+
+import { type IntentResolver, LayoutAction, useIntentDispatcher, useIntentResolver } from '@dxos/app-framework';
+import { debounce } from '@dxos/async';
+import { fullyQualifiedId } from '@dxos/react-client/echo';
+import { type DxGridElement, type DxGridPosition } from '@dxos/react-ui-grid';
+
+import { useSheetContext } from '../components';
+import { type CellAddress, type CompleteCellRange, inRange } from '../defs';
+import { SHEET_PLUGIN } from '../meta';
+
+export const completeCellRangeToThreadCursor = (range: CompleteCellRange): string => {
+  return `${range.from.col},${range.from.row},${range.to.col},${range.to.row}`;
+};
+
+export const parseThreadAnchorAsCellRange = (cursor: string): CompleteCellRange | null => {
+  const coords = cursor.split(',');
+  if (coords.length !== 4) {
+    return null;
+  } else {
+    const [fromCol, fromRow, toCol, toRow] = coords;
+    return {
+      from: { col: parseInt(fromCol), row: parseInt(fromRow) },
+      to: { col: parseInt(toCol), row: parseInt(toRow) },
+    };
+  }
+};
+
+export const useUpdateFocusedCellOnThreadSelection = (grid: MutableRefObject<DxGridElement | null>) => {
+  const { model, setActiveRefs } = useSheetContext();
+  const handleScrollIntoView: IntentResolver = useCallback(
+    ({ action, data }) => {
+      switch (action) {
+        case LayoutAction.SCROLL_INTO_VIEW: {
+          if (!data?.id || data?.cursor === undefined || data?.id !== fullyQualifiedId(model.sheet)) {
+            return;
+          }
+          setActiveRefs(data.thread);
+          // TODO(Zan): Everywhere we refer to the cursor in a thread context should change to `anchor`.
+          const range = parseThreadAnchorAsCellRange(data.cursor);
+          range && grid.current?.setFocus({ ...range.to, plane: 'grid' }, true);
+        }
+      }
+    },
+    [model.sheet, setActiveRefs],
+  );
+
+  useIntentResolver(SHEET_PLUGIN, handleScrollIntoView);
```

Comment:
i think useIntentResolver should be removed since it obfuscates what's actually happening. i wrote this and it was intended to be short lived.

better to replace the callback above with:

```ts
  const { registerResolver } = useIntent();
  useEffect(() => {
    return registerResolver()
  }, []);
```

same pattern in the 3 other places.

Reply:
The scope of this PR might not be great for this refactor as I’m not familiar with this pattern and just moved it to a new file.

Reply:
I do think Sheet’s understanding of ranges (using `to` and `from`) are meaningfully distinct from `dx-grid`’s (using `start` and `end`), since either may or may not sort one coordinate relative to the other…